### DISCUSSION TITLE
Vote for storage transformations

### DISCUSSION THREAD

**Discussion by inian** [supabase/supabase#10918] [Feature Requests]
This issue is for tracking requests/demand for different image transformations. Give a comment an upvote if you are interested in a particular transformation.

Please refrain from adding comments unless you are adding a transformation which isn't already listed.

**Comment by inian** [comment]
Image Rotation

**Comment by inian** [comment]
Blur

**Reply by 0xflotus** [reply]
What kind of Blur are we talking about?

In [imagemagick](https://imagemagick.org/script/examples.php) there are some different kinds of blurring an image possible like Gaussian Blur, Motional Blur or Rotational Blur...

And depending of your needs, it is sometimes worth to consider to use something like [Blurhash](https://github.com/woltapp/blurhash), so you only need to transfer a small hash and not a big bytestream of blurred image data.

**Reply by KhaledGabr** [reply]
couldn't we blur images using CSS ?

**Reply by henryruhs** [reply]
blur of images is important if you want to anonymize a profile picture for example

**Reply by remorses** [reply]
a small blur image could be returned together with the image aspect ratio after you upload an image with the sdk so you can store it in the database and use it as loading placeholder

**Comment by inian** [comment]
Quality ✅

**Reply by segheysens** [reply]
Is the main idea here downsampling for compressing image file sizes?

**Reply by inian** [reply]
yup

**Reply by KhaledGabr** [reply]
Both lossless and lossy Image compression can be used for reducing file size. But I agree. it's pretty much the most important thing after resizing. People want their images to load quickly and we can't expect them to upload perfectly compressed images.

**Reply by inian** [reply]
This has been launched with [storage v3](https://supabase.com/blog/storage-v3-resumable-uploads#more-image-transformations-options)!  Docs for setting the quality are [here](https://supabase.com/docs/reference/javascript/storage-from-download).

**Comment by james2doyle** [comment]
Smart Crop

**Reply by KhaledGabr** [reply]
Is the idea here to set the focal point so that the area of interest in an image doesn't get cropped out ?

**Comment by madeleineostoja** [comment]
Format conversion ✅

**Reply by madeleineostoja** [reply]
As an MVP version of the feature even just an `auto` option to webp/avif, for performant responsive images along with resizing

**Reply by segheysens** [reply]
Support for other filetypes would be awesome! I'm building functionality for users to be able to upload TXT, Word, and PDF files and want to standardize to store PDF copies. Currently testing out [this Cloudinary add-on](https://cloudinary.com/documentation/aspose_document_conversion_addon), but might rewire things to use Supabase Storage + webhooks for now. I'd also like to generate a thumbnail preview image of the documents on upload.

**Reply by madeleineostoja** [reply]
Afaik the transformations API is only for images for now (and forseeable future?), I imagine things like converting txt/word/pdf and generating thumbnails of them would be beyond the scope of this api, happy to be corrected though. 

I ran a quick spike with the new API compared to the exact same srcsets using Imgix (w/ AVIF and 75% quality filter) and Supabase assets were still ~3-4x larger.

**Reply by inian** [reply]
we will be adding webp support soon :) We will need to benchmark AVIF encoders since they are notoriously slow, but once we do that AVIF should land soon-ish too.

**Reply by inian** [reply]
Images going through the render endpoint are automatically converted to WebP now as of [Storage V3](https://supabase.com/blog/storage-v3-resumable-uploads#more-image-transformations-options)

**Reply by madeleineostoja** [reply]
Storage v3 looks awesome! How can I get on the early access list?

**Reply by inian** [reply]
Webp and quality filters are available for all projects. Here are the [docs](https://supabase.com/docs/reference/javascript/storage-from-download).

Please send a support ticket via https://app.supabase.com/support/new for resumable uploads

**Reply by gustavopch** [reply]
SVG -> PNG would be great to create dynamic images.

**Comment by 0xflotus** [comment]
Grayscale

**Comment by Macko-pp** [comment]
Invert Colors

**Reply by vincentjflorio** [reply]
In the meantime, this is also a CSS supported feature under the filters property.

https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/invert

**Comment by Macko-pp** [comment]
Hue shifting

**Comment by psteinroe** [comment]
not image, but video: thumbnail generation

**Comment by gehlert** [comment]
Watermarks

**Reply by mrkpatchaa** [reply]
Same here. I tried to play a little bit with img_proxy config in self hosted but it's not working.
```
      IMGPROXY_WATERMARK_PATH: /home/watermark_small.png
      IMGPROXY_PRESETS: default=watermark:1
      IMGPROXY_LOG_LEVEL: "debug"
    volumes:
      - ./volumes/storage:/var/lib/storage:z
      - ./volumes/images:/home:cached
```

**Reply by mrkpatchaa** [reply]
My bad. I made it work. Seems like image was cached for a time. 
`IMGPROXY_PRESETS: default=watermark:1:sowe:10:10:0.4`
More info in possible configuration of watermark here https://docs.imgproxy.net/watermark?id=watermarking-an-image

**Reply by iamgoddog** [reply]
This would be a nice feature.

**Reply by kkukshtel** [reply]
Would also really love this! Basically, just image compositing. I'm working on a product where users would want to watermark their images for clients

**Comment by marcusarnfast** [comment]
### Rounded corners

Would be nice if we could round the corners of images. Especially considering the limitations of certain email clients like MS Outlook, which do not support CSS properties for rounded borders directly.

Here is how Cloudinary does it. 
[https://cloudinary.com/cookbook/applying_rounded_corners_to_an_image](url)

**Comment by skikrumb-ryan** [comment]
Trim https://docs.imgix.com/apis/rendering/trim

**Comment by mmarchois** [comment]
Video compression
